### PR TITLE
HPCC-13241 Add support for ESDL file extension

### PR DESCRIPTION
**Pull request by rpastrana**
- Continue to support ecm extension
- Update usage statements
- #define relevant extensions

Signed-off-by: rpastrana rodrigo.pastrana@lexisnexis.com

### CODE REVIEW

**Comment by rpastrana** [comment]
Currently targeting 5.4, but this might be needed earlier if possible.

**Comment by rpastrana** [comment]
@afishbeck please review.

File: tools/esdlcmd/esdl-publish.cpp [hunk 1]
```diff
@@ -160,11 +160,11 @@ class EsdlPublishCmd : public EsdlPublishCmdCommon
     void usage()
     {
         printf("\nUsage:\n\n"
-                "esdl publish <servicename> <filename.ecm> [command options]\n\n"
-                "   <servicename>        The ESDL defined ESP service to publish\n"
-                "   <filename.ecm>       The ESDL file containing service definition\n"
+                "esdl publish <servicename> <filename.(ecm|esdl)> [command options]\n\n"
```
Comment (afishbeck) [inline tools/esdlcmd/esdl-publish.cpp]: It's fine to document this way, but as far as the user specified file I would think that any extension would be supported?  I.e. ".xxyyzz" should work?  We just don't have to advertise it.

Comment (afishbeck) [inline tools/esdlcmd/esdl-publish.cpp]: It's more the include files that should look for either .ecm or .esdl.

File: tools/esdlcmd/esdl2ecl.cpp [hunk 2]
```diff
@@ -282,6 +282,7 @@ class Esdl2EclCmd : public EsdlConvertCmd
         unsigned start = msTick();
         EsdlIndexedPropertyTrees trees;
         trees.loadFile(srcPath.str(), srcName.str(), srcExt.str(), NULL, optGenerateAllIncludes);
+        //trees.loadFile(srcPath.str(), srcName.str(), srcExt.str(), NULL, true);
```
Comment (afishbeck) [inline tools/esdlcmd/esdl2ecl.cpp]: remove commented out code?  Maybe a comment on the line above if something needs explaining

File: tools/esdlcomp/esdlcomp.cpp [hunk 3]
```diff
@@ -1109,8 +1107,22 @@ ESDLcompiler::ESDLcompiler(const char * sourceFile, bool generatefile, const cha
...
+        fprintf(stderr, "Could not load: %s...\n",sourceFile);
+        StringBuffer alternateExtFilename(srcDir);
+        alternateExtFilename.append(name.str());
+
+        if (stricmp(ext.str(), ESDL_FILE_EXTENTION)==0)
+            alternateExtFilename.append(LEGACY_FILE_EXTENTION);
+        else
+            alternateExtFilename.append(ESDL_FILE_EXTENTION);
+
+        fprintf(stderr, "Attempting to re-load with alternate extension: %s\n",alternateExtFilename.str());
```
Comment (afishbeck) [inline tools/esdlcomp/esdlcomp.cpp]: I think we need to know if we got here via include or this is the top level file specified on command line.

If this is an include we should check both extensions silently.
If this is the file specified on command line we should use exactly what the user specified.

In the future we may want to allow the full name to be specified in the Include statement.

File: tools/esdlcmd/esdl2ecl.cpp (hunk 2 above)
Comment (rpastrana) [inline tools/esdlcmd/esdl2ecl.cpp]: That one snuck in

File: tools/esdlcmd/esdl-publish.cpp (hunk 1 above)
Comment (rpastrana) [inline tools/esdlcmd/esdl-publish.cpp]: this particular command actually accepts esdl or esxdl, we could open the file and inspect the content to determine if the file contains esdl which needs to be processed to esxdl first, or if it's xml which it attempts to load regardless of extension. Right now we use the ecm/esdl extension to determine if the content needs to be processed (the usage needs to reflect this)

**Comment by rpastrana** [comment]
@afishbeck please review latest commit. Also, this is currently targeting 5.4.0 but I think this should be included in 5.2.2, what do you think?

**Comment by afishbeck** [comment]
@rpastrana Except that commented out line, looks good.

**Pull request closed without merging**